### PR TITLE
auto_optim enhancement

### LINKED ISSUE #2168: Feature Request: add an additional argument to auto_optim to allow for gradient accumulation
When we use horovod backend and perform gradient accumulation, we get the following error:
`AssertionError: Gradients were computed more than backward_passes_per_step times before call to step(). Increase backward_passes_per_step to accumulate gradients
locally.`

Thus, we need to change the default argument `backward_passes_per_step` of `horovod.DistributedOptimizer` to enable gradient accumulation in the distributed setting.  To do so, we can add this argument to `ignite.distributed.auto_optim`.

### PR DESCRIPTION
**Pull request by Chandan-h-509**
Fixes #2168

Description: Add kwargs to the auto_optim argument and pass it to DistributedOptimizer .

Check list:

- [x] New tests are added (if a new feature is added)
- [x] New doc strings: description and/or example code are in RST format
- [x] Documentation is updated (if required)

### CODE REVIEW

**Comment by sdesrozis** [comment]
@Chandan-h-509 Thank you ! 

It would be nice to have a test of this new feature.

**Comment by Chandan-h-509** [comment]
@sdesrozis I have added the test case.
Please review.
Thank you!

File: ignite/distributed/auto.py [hunk 1]
```diff
@@ -234,7 +234,7 @@ def auto_model(model: nn.Module, sync_bn: bool = False, **kwargs: Any) -> nn.Mod
     return model
 
 
-def auto_optim(optimizer: Optimizer) -> Optimizer:
+def auto_optim(optimizer: Optimizer, **kwargs: Any) -> Optimizer:
```
Comment (sdesrozis) [inline ignite/distributed/auto.py:237]: A tag `.. versionchanged:: 0.5.0` must be added in docstring. Please look the docstrings above for an example.

**Comment by sdesrozis** [comment]
@Chandan-h-509 a minor comment to address and a format error to fix. See https://github.com/pytorch/ignite/pull/2169/checks?check_run_id=3389112372

**Comment by Chandan-h-509** [comment]
@sdesrozis   I have made the final changes.
Please review.
Thank you.

**Review by sdesrozis — APPROVED** [review]
@Chandan-h-509 Thank you ! LGTM !

File: tests/ignite/distributed/test_auto.py [hunk 2]
```diff
@@ -164,7 +164,7 @@ def _test_auto_model_optimizer(ws, device):
     # Test auto_optim
     bnd = idist.backend()
     optimizer = optim.SGD(model.parameters(), lr=0.01)
-    optimizer = auto_optim(optimizer)
+    optimizer = auto_optim(optimizer, backward_passes_per_step=2)
```
Comment (vfdev-5) [inline tests/ignite/distributed/test_auto.py:167]: Let's create another specific test instead of overriding original one. Please, revert this change and add the following:
```python
if idist.has_hvd_support and bnd in ("horovod",):
    backward_passes_per_step = 2
    optimizer = optim.SGD(model.parameters(), lr=0.01)
    optimizer = auto_optim(optimizer, backward_passes_per_step=backward_passes_per_step)    
    assert isinstance(optimizer, optim.SGD) and hasattr(optimizer, "backward_passes_per_step")
    assert optimizer.backward_passes_per_step == backward_passes_per_step
```

**Comment by vfdev-5** [comment]
cc @sandylaker for visibility

File: tests/ignite/distributed/test_auto.py (hunk 2 above)
Comment (Chandan-h-509) [reply tests/ignite/distributed/test_auto.py:167]: Sure.. I'll do that.

**Comment by Chandan-h-509** [comment]
@vfdev-5 I have made the new change as requested.
Kindly review.

File: tests/ignite/distributed/test_auto.py [hunk 3]
```diff
@@ -171,6 +171,12 @@ def _test_auto_model_optimizer(ws, device):
         assert isinstance(optimizer, optim.SGD) and hasattr(optimizer, "_allreduce_grad_async")
     else:
         assert isinstance(optimizer, optim.SGD) and not hasattr(optimizer, "wrapped_optimizer")
+    if idist.has_hvd_support and bnd in ("horovod",):
```
Comment (vfdev-5) [inline tests/ignite/distributed/test_auto.py:174]:
```suggestion

    if idist.has_hvd_support and bnd in ("horovod",):
```
